pandas/core/ops.py

```diff
@@ -2272,10 +2272,10 @@ def _cast_sparse_series_op(left, right, opname):
     # TODO: This should be moved to the array?
     if is_integer_dtype(left) and is_integer_dtype(right):
         # series coerces to float64 if result should have NaN/inf
-        if opname in ('floordiv', 'mod') and (right.values == 0).any():
+        if opname in ('floordiv', 'mod') and (right.to_dense() == 0).any():
```

Comment:
should we not be using ``np.asarry``? generally rather than ``.to_dense()``?

Reply:
Both are equivalent (although `to_dense` actually does a bit less as it specified the dtype and `asarray` does some inference (not sure for that difference though)).